pkg/signup/service/signup_service.go

```diff
@@ -589,33 +595,49 @@ func (s *ServiceImpl) DoGetUserSignupFromIdentifier(provider ResourceProvider, u
 
 // UpdateUserSignup is used to update the provided UserSignup resource, and returning the updated resource
 func (s *ServiceImpl) UpdateUserSignup(userSignup *toolchainv1alpha1.UserSignup) (*toolchainv1alpha1.UserSignup, error) {
-	userSignup, err := s.CRTClient().V1Alpha1().UserSignups().Update(userSignup)
-	if err != nil {
+	if err := s.Update(gocontext.TODO(), userSignup); err != nil {
 		return nil, err
 	}
 
 	return userSignup, nil
 }
 
+var (
+	md5Matcher = regexp.MustCompile("(?i)[a-f0-9]{32}$")
+)
+
 // PhoneNumberAlreadyInUse checks if the phone number has been banned. If so, return
-// an internal server error. If not, check if an active (non-deactivated) UserSignup with a different userID and username
+// an internal server error. If not, check if an approved UserSignup with a different userID and username
 // and email address exists. If so, return an internal server error. Otherwise, return without error.
 // Either the actual phone number, or the md5 hash of the phone number may be provided here.
 func (s *ServiceImpl) PhoneNumberAlreadyInUse(userID, username, phoneNumberOrHash string) error {
-	bannedUserList, err := s.CRTClient().V1Alpha1().BannedUsers().ListByPhoneNumberOrHash(phoneNumberOrHash)
-	if err != nil {
+	labelValue := hash.EncodeString(phoneNumberOrHash)
+	if md5Matcher.Match([]byte(phoneNumberOrHash)) {
```

Comment:
is strange that the md5 package doesn't have a function for identifying if a string is md5 hash. 

Also should we add a unit tests for this case ?

Reply:
I took a quick glance and I didn't find anything. But TBH, this is not really needed - it can be done without checking the format of the provided value, it just requires second function/method which is hard with the current setup. Can be done later - let me create a story for it.

Reply:
https://issues.redhat.com/browse/KUBESAW-200

Reply:
Thanks for creating the Jira, and yeah it would be nicer if we could remove that regex matcher :)